frontend/actions.js

```diff
@@ -98,58 +89,149 @@ export default ({setState, getState}) => {
     displayAboutOverlay: !state.displayAboutOverlay,
   })
 
-  const inputAddress = (state, e) => ({
-    sendAddress: e.target.value,
+  const clearSendAmountValidation = (state) => ({
+    sendAmount: Object.assign(state.sendAmount, {validationMsg: ''}),
+  })
+
+  const clearSendAddressValidation = (state) => ({
+    sendAddress: Object.assign(state.sendAddress, {validationMsg: ''}),
   })
 
-  // is being called debounced, thus we need to getState
-  // TODO maybe it'd be good practice to only use getState and avoid this problem?
+  const validateSendAddress = (state) => {
+    if (!Cardano.isValidAddress(state.sendAddress.fieldValue)) {
+      setState({
+        sendAddress: Object.assign(state.sendAddress, {validationMsg: 'Invalid address'}),
+      })
+      return false
+    }
+    return true
+  }
+
+  const validateSendAmount = (state) => {
+    if (isNaN(parseFloat(state.sendAmount.fieldValue))) {
+      setState({
+        sendAmount: Object.assign(state.sendAmount, {
+          validationMsg: 'Invalid format: Amount has to be a number',
+        }),
+      })
+      return false
+    }
+    if (parseFloat(state.sendAmount.fieldValue).toFixed(6) <= 0) {
+      setState({
+        sendAmount: Object.assign(state.sendAmount, {
+          validationMsg: 'Invalid format: Amount has to be a positive number',
+        }),
+      })
+      return false
+    }
+    if (parseFloat(state.sendAmount.fieldValue).toFixed(6) > state.balance / 1000000) {
+      setState({
+        sendAmount: Object.assign(state.sendAmount, {
+          validationMsg: `Insufficient funds. Your balance is ${state.balance / 1000000} ADA.`,
+        }),
+      })
+      return false
+    }
+    return true
+  }
+
+  const validateFee = (state) => {
+    if (parseFloat(state.transactionFee) < 0) {
+      setState({
+        transactionFee: -state.transactionFee,
+        sendAmount: Object.assign(state.sendAmount, {
```

Comment:
you can return instead of using setState, it's nicer, keep setState for async functions